lib/core/test/unit/Cardano/Pool/MetricsSpec.hs

```diff
@@ -101,18 +108,49 @@ prop_combineIsLeftBiased mStake mProd mPerf =
 prop_performancesBounded01
     :: Map PoolId (Quantity "lovelace" Word64)
     -> Map PoolId (Quantity "block" Word64)
+    -> (NonNegative Int)
     -> Property
-prop_performancesBounded01 mStake mProd =
+prop_performancesBounded01 mStake mProd (NonNegative emptySlots) =
     all (between 0 1) performances
     & counterexample (show performances)
     & classify (all (== 0) performances) "all null"
   where
     performances :: [Double]
-    performances = Map.elems $ calculatePerformance mStake mProd
+    performances = Map.elems $ calculatePerformance slots mStake mProd
+
+    slots :: Int
+    slots = emptySlots +
+        fromIntegral (Map.foldl (\y (Quantity x) -> (y + x)) 0 mProd)
 
     between :: Ord a => a -> a -> a -> Bool
     between inf sup x = x >= inf && x <= sup
 
+
+performanceGoldens :: Spec
+performanceGoldens = do
+    it "50% stake, producing 4/8 blocks => performance=1" $ do
```

Comment:
I actually made this a property test for a moment. That property test was however basically a reimplementation of the `calculatePerformance`-function, so I went back to the goldens.

The goldens are also very easy to read as a human.